### DISCUSSION TITLE
Different results for the same video in different runs

### DISCUSSION THREAD

**Discussion by mennatullahnoaman** [stella-cv/stella_vslam#397] [Q&A] [answered]
Hello all,

First I would thank you for the great work.

Second I want to ask why different runs of example run_video_slam.cc on the same video get different results ?
I follow the code and set all fixed_seed with true by :
`sed -i -- 's/use_fixed_seed = false/use_fixed_seed = true/g' src/stella_vslam/*/*.h` 
so modified files :   initializer.cc , loop_detector.cc and  relocalizer.cc 
and also add use_fixed_seed: true in the config.yaml file for loopdetector and Initializer

#=========================#
# LoopDetector Parameters #
#=========================#
LoopDetector:
  enabled: true
  reject_by_graph_distance: true
  min_distance_on_graph: 50
  use_fixed_seed: true
#========================#
# Initializer Parameters #
#========================#
Initializer:
  use_fixed_seed: true  
but still there is small different between two runs .. **Can you help me figure out what is the reason for this difference?**

Third I would like to know the example configs file parameter values is it set random or it is based on experiments ?

Again, Thanks for the great efforts.

**Comment by ymd-stella** [comment, marked as answer]
Check DETERMINISTIC compile option.

https://github.com/stella-cv/stella_vslam/blob/b55ba925c40e708a1cd2181038faf157e6984a7e/src/stella_vslam/CMakeLists.txt#L148
